worker/src/queue.rs

```diff
@@ -1,124 +1,109 @@
-use std::marker::PhantomData;
-
 use crate::{env::EnvBinding, Date, Error, Result};
 use js_sys::Array;
 use serde::{de::DeserializeOwned, Serialize};
 use wasm_bindgen::{prelude::*, JsCast};
 use wasm_bindgen_futures::JsFuture;
-use worker_sys::{MessageBatch as MessageBatchSys, Queue as EdgeQueue};
-
-static BODY_KEY_STR: &str = "body";
-static ID_KEY_STR: &str = "id";
-static TIMESTAMP_KEY_STR: &str = "timestamp";
+use worker_sys::{Message as MessageSys, MessageBatch as MessageBatchSys, Queue as EdgeQueue};
 
-pub struct MessageBatch<T> {
+/// A batch of messages that are sent to a consumer Worker.
+pub struct MessageBatch {
     inner: MessageBatchSys,
     messages: Array,
-    data: PhantomData<T>,
-    timestamp_key: JsValue,
-    body_key: JsValue,
-    id_key: JsValue,
 }
 
-impl<T> MessageBatch<T> {
+impl MessageBatch {
     pub fn new(message_batch_sys: MessageBatchSys) -> Self {
-        let timestamp_key = JsValue::from_str(TIMESTAMP_KEY_STR);
-        let body_key = JsValue::from_str(BODY_KEY_STR);
-        let id_key = JsValue::from_str(ID_KEY_STR);
         Self {
             messages: message_batch_sys.messages(),
             inner: message_batch_sys,
-            data: PhantomData,
-            timestamp_key,
-            body_key,
-            id_key,
         }
     }
 }
 
-pub struct Message<T> {
-    pub body: T,
-    pub timestamp: Date,
-    pub id: String,
-}
-
-impl<T> MessageBatch<T> {
+impl MessageBatch {
     /// The name of the Queue that belongs to this batch.
     pub fn queue(&self) -> String {
-        self.inner.queue()
+        self.inner.queue().into()
     }
 
     /// Marks every message to be retried in the next batch.
     pub fn retry_all(&self) {
         self.inner.retry_all();
     }
 
-    /// Iterator that deserializes messages in the message batch. Ordering of messages is not guaranteed.
-    pub fn iter(&self) -> MessageIter<'_, T>
-    where
-        T: DeserializeOwned,
-    {
+    /// Marks every message acknowledged in the batch.
+    pub fn ack_all(&self) {
+        self.inner.ack_all();
+    }
+
+    /// Iterator for messages in the message batch. Ordering of messages is not guaranteed.
+    pub fn iter(&self) -> MessageIter<'_> {
         MessageIter {
             range: 0..self.messages.length(),
             array: &self.messages,
-            timestamp_key: &self.timestamp_key,
-            body_key: &self.body_key,
-            id_key: &self.id_key,
-            data: PhantomData,
         }
     }
+}
+
+/// A message that is sent to a consumer Worker.
+pub struct Message {
+    inner: MessageSys,
+}
 
-    /// An array of messages in the batch. Ordering of messages is not guaranteed.
-    pub fn messages(&self) -> Result<Vec<Message<T>>>
+impl Message {
+    pub fn new(message_sys: MessageSys) -> Self {
```

Comment:
This should probably be an `impl From<MessageSys>`

Reply:
Done